torch/_inductor/fx_passes/quantization.py

```diff
@@ -1150,22 +1150,100 @@ def woq(match: Match, *args, **kwargs):
     return woq
 
 
-def _register_woq_mm_int8():
-    # torch.nn.functional.linear(x, weight.to(dtype=x.dtype)) * scales
+def _register_woq_mm_int8_pattern1():
+    # F.linear(x, weight.to(dtype=x.dtype)) * scales
     _woq_pattern = CallFunction(
         aten.mul.Tensor,
         CallFunction(
             aten.reshape.default,
             CallFunction(
-                aten.mm.default,
-                CallFunction(aten.reshape.default, KeywordArg("x"), Arg()),
+                prims.convert_element_type.default,
                 CallFunction(
-                    aten.permute.default,
+                    aten.sum.dim_IntList,
                     CallFunction(
-                        prims.convert_element_type.default, KeywordArg("weight"), Arg()
+                        aten.mul.Tensor,
+                        CallFunction(
+                            aten.unsqueeze.default,
+                            CallFunction(
+                                prims.convert_element_type.default,
+                                CallFunction(
+                                    aten.reshape.default, KeywordArg("x"), Arg()
+                                ),
+                                Arg(),
+                            ),
+                            Arg(),
+                        ),
+                        CallFunction(
+                            aten.unsqueeze.default,
+                            CallFunction(
+                                prims.convert_element_type.default,
+                                CallFunction(
+                                    aten.permute.default,
+                                    CallFunction(
+                                        prims.convert_element_type.default,
+                                        KeywordArg("weight"),
+                                        Arg(),
+                                    ),
+                                    Arg(),
+                                ),
+                                Arg(),
+                            ),
+                            Arg(),
+                        ),
                     ),
                     Arg(),
                 ),
+                Arg(),
+            ),
+            Arg(),
+        ),
+        KeywordArg("scales"),
+    )
+    _register_woq_lowering(_woq_pattern, aten._weight_int8pack_mm.default, aten.reshape)
+
+
+def _register_woq_mm_int8_pattern2():
+    # F.linear(x, weight.to(dtype=x.dtype)) * scales
```

Comment:
Please add comments to explain the differences of pattern1 and pattern2 and why we need both.

Reply:
Thanks and added.